grant-project/zcash-grant-system
# Address TODOs

Closes #326.

⚠️  If you disagree with any of the removed but unaddressed TODOs, please let me know in comments.

### What this does
- addresses non-webargs related TODOs I've used my judgment to attempt to determine what was needed in each case
- removed some comments that I deemed unnecessary
- converted some comments to NOTEs
- addressed several TODOs:
    - add validation to be `user` model, hook it into create
    - improved `test_invite_api` tests ( a very little bit)
    - capture error messages in some of the reducers
    - upgraded `react-helmet` and enabled `HeaderDetails` component
    - added `staking=True` filter to the query to get active staking contribution for `get_staking_contribution` function of proposal model
    - fix remaining staking contribution calculation (was using `contributed` which filters out staking proposals)

### Notes
- tests are needed for partial staking

### Testing
- update yarn deps to get the latest helmet
- helmet
    - visit the **homepage** (full reload), inspect the very first HTML received in the networks tab, it should have an `og:description` like the below:
    ![Screen Shot 2019-03-13 at 1 04 54 PM](https://user-images.githubusercontent.com/11430954/54303283-a22d5d80-4590-11e9-94cf-ee7b03ae07fa.png)
    - visit a **profile**, check that it has profile-specific og and twitter meta tags:
    ![Screen Shot 2019-03-13 at 1 09 29 PM](https://user-images.githubusercontent.com/11430954/54303568-42838200-4591-11e9-8ae0-b96e92b4ad94.png)
- `get_staking_contribution` 
    - create a proposal and stake it for less than the required amount (use admin if you like)
    - go to profile - pending tab, click 'stake' button for proposal, the amount should be the full stake amount minus the amount already staked
- user validation
    - attempt to create a user with a `title` or `display_name` over 255 chars long (below is 260 chars)
    ```asdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfgasdfg```
    - you should see something like the following:
    ![Screen Shot 2019-03-13 at 2 01 47 PM](https://user-images.githubusercontent.com/11430954/54307024-aeb5b400-4598-11e9-931c-d2e20d9514e2.png)
